ank/doc/swdesign/README.md

```diff
@@ -206,6 +210,102 @@ Needs:
 - impl
 - itest
 
+#### CLI supports PEM file paths as cli arguments for mTLS
+`swdd~cli-supports-pem-file-paths-as-cli-arguments~1`
+
+Status: approved
+
+The Ankaios CLI shall support the following cli arguments all of which contain a file path to a PEM file:
+* `--ca_pem`: file path to the root certificate authority
+* `--crt_pem`: file path to the certificate
+* `--key_pem`: file path to the key
+
+Comment:
+The PEM file format is a text-based, human-readable format. PEM files can contain certificates, private keys, public keys and even certificate chains, making them more versatile compared to other formats.
+
+Rationale:
+The Ankaios CLI uses the PEM files to establish a mutual transport layer security (mTLS) communication.
+
+Tags:
+- CliCommands
+
+Needs:
+- impl
+- stest
+
+#### CLI supports cli argument for insecure communication
+`swdd~cli-supports-cli-argument-for-insecure-communication~1`
+
+Status: approved
+
+The Ankaios CLI shall support the cli argument `--insecure` with short alias `-k`.
+
+Comment:
+The Ankaios CLI supports an insecure communication channel to the Ankaios server. Communicating over insecure channels is convenient during an evaluation or a development phase.
+
+Tags:
+- CliCommands
+
+Needs:
+- impl
+- stest
+
+#### CLI provides file paths for mTLS to communication middleware
+`swdd~cli-provides-file-paths-to-communication-middleware~1`
+
+Status: approved
+
+When the file paths of the root certificate, the CLI certificate and the CLI key is provided upon startup, then the Ankaios CLI shall provide each of these file paths to the communication middleware.
+
+Comment:
+The communication middleware parses the content of the files.
+
+Rationale:
+The communication middleware is responsible for establishing a secure communication channel.
+
+Tags:
+- AgentManager
+- CommunicationMiddleware
+
+Needs:
+- impl
+
+#### CLI establishes insecure connection on provided insecure cli argument
+`swdd~cli-establishes-insecure-communication-based-on-provided-insecure-cli-argument~1`
+
+Status: approved
+
+When the user runs the Ankaios CLI with the arguments `-k` or `--insecure` and the Ankaios CLI initializes the communication middleware, then the Ankaios CLI shall instruct the communication middleware to use an insecure communication channel.
+
+Comment:
+Communicating over insecure channels is convenient during an evaluation or a development phase.
+
+Tags:
+- CliCommands
+- CommunicationMiddleware
```

Comment:
We don't need to tag the CommunicationMiddleware. It is an external component and tagging it does not bring any benefit.
```suggestion
```